### PR TITLE
Cli / get all tasks

### LINKED ISSUE #127: Create Catalog CLI
It is necessary to obtain information about the tasks contained in the DB to monitor the progress of the processing, if the admin does not have access via GUI it is important to have a CLI to resolve this.

### PR DESCRIPTION
**Pull request by thiagoyeds**
Resolved #127

### CODE REVIEW

File: bin/get-all-tasks [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  echo "Usage: $0 USER_EMAIL USER_PASSWORD DISPATCHER_ADDRESS"
+  exit 1
+fi
+
+# args
+readonly USER_EMAIL=$1
+readonly USER_PASSWORD=$2
+readonly DISPATCHER_ADDRESS=$3
+
+ALL_TASKS=`curl --silent --request GET "${DISPATCHER_ADDRESS}/processings" --header "Content-Type: text/plain"  --header "userEmail: $USER_EMAIL"  --header "userPass: $USER_PASSWORD"`
```
Comment (wesleymonte) [inline bin/get-all-tasks]: Use $() instead backticks. Backticks have been deprecated.

File: bin/get-all-tasks [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+JSON_LENGTH=$((`echo $ALL_TASKS | jq 'length' $JSON_PATH` - 1))
```
Comment (wesleymonte) [inline bin/get-all-tasks]: $JSON_PATH is not used by the jq command. It is empty.
Use $() instead backticks. Backticks have been deprecated.

File: bin/get-all-tasks [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+declare -a FILTERS_ARRAY=("taskId" "dataset" "region" "inputGatheringTag" "inputPreprocessingTag" "algorithmExecutionTag" "state" "creationTime" "updateTime")
+FILTERS_ARRAY_LENGTH=$((${#FILTERS_ARRAY[@]} - 1))
+
+echo "${FILTERS_ARRAY[@]}"
+
+for json_index in `seq 0 $JSON_LENGTH`; do
```
Comment (wesleymonte) [inline bin/get-all-tasks]: Use $() instead backticks.

**Review by wesleymonte — CHANGES_REQUESTED** [review]

File: bin/get-all-tasks (hunk 3 above)
Comment (thiagoyeds) [reply bin/get-all-tasks]: Done

File: bin/get-all-tasks (hunk 2 above)
Comment (thiagoyeds) [reply bin/get-all-tasks]: Done

File: bin/get-all-tasks (hunk 1 above)
Comment (thiagoyeds) [reply bin/get-all-tasks]: Done

File: bin/get-all-tasks [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+## This script get all tasks for SAPS interacting with Dispatcher backend.
```
Comment (thiagomanel) [inline bin/get-all-tasks:3]: Gives an example of the expected output

File: bin/get-all-tasks [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+ALL_TASKS=$(curl --silent --request GET "${DISPATCHER_ADDRESS}/processings" --header "Content-Type: text/plain"  --header "userEmail: $USER_EMAIL"  --header "userPass: $USER_PASSWORD")
+JSON_LENGTH=$(($(echo $ALL_TASKS | jq 'length') - 1))
+
+declare -a FILTERS_ARRAY=("taskId" "dataset" "region" "inputGatheringTag" "inputPreprocessingTag" "algorithmExecutionTag" "state" "creationTime" "updateTime")
+FILTERS_ARRAY_LENGTH=$((${#FILTERS_ARRAY[@]} - 1))
+
+echo "${FILTERS_ARRAY[@]}"
+
+for json_index in $(seq 0 $JSON_LENGTH); do
```
Comment (thiagomanel) [inline bin/get-all-tasks:34]: this loop is complex. add a comment explaining what it does

File: bin/get-all-tasks [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+fi
+
+# args
+readonly USER_EMAIL=$1
+readonly USER_PASSWORD=$2
+readonly DISPATCHER_ADDRESS=$3
+
+ALL_TASKS=$(curl --silent --request GET "${DISPATCHER_ADDRESS}/processings" --header "Content-Type: text/plain"  --header "userEmail: $USER_EMAIL"  --header "userPass: $USER_PASSWORD")
+JSON_LENGTH=$(($(echo $ALL_TASKS | jq 'length') - 1))
+
```
Comment (thiagomanel) [inline bin/get-all-tasks:25]: I'm wondering if we should list all these properties by default. Or, if we should omit the tag stuff. Maybe, the tag columns (gathering, preproc, proc) should appear when the user gives a --tags option.

what do you think @wesleymonte @thiagoyeds and @raonismaneoto?

**Review by thiagomanel — CHANGES_REQUESTED** [review]
two things:
1. this code is complex. we need to add a couple of comment lines
2. Maybe we should allow this script to take some options

File: bin/get-all-tasks (hunk 4 above)
Comment (thiagoyeds) [reply bin/get-all-tasks:3]: Done

File: bin/get-all-tasks (hunk 5 above)
Comment (thiagoyeds) [reply bin/get-all-tasks:34]: Done

**Review by thiagomanel — APPROVED** [review]
lgtm

**Pull request merged**